### PR TITLE
prevent before divideByZero + extend coverage up to hundred + add sta…

### PR DESCRIPTION
**Pull request by MockingMagician**
- early prevent for divideByZero
- extend coverage up to hundred
- add static analysis
- add some package defined exception
- add some composer scripts related to tests

### CODE REVIEW

**Comment by romainnorberg** [comment]
Hy @MockingMagician Nice, thanks 🆙 . Could you rebase this with the current master ? I pushed some modifications _(including the static constructor)_ before seeing your PR

File: src/Exception/DecimalException.php [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+/*
+ * This file is part of the Residue package.
+ * (c) Romain Norberg <romainnorberg@gmail.com>
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Romainnorberg\Residue\Exception;
+
+use Exception;
```
Comment (romainnorberg) [inline src/Exception/DecimalException.php]: unused

File: src/Exception/DivideByZeroException.php [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+/*
+ * This file is part of the Residue package.
+ * (c) Romain Norberg <romainnorberg@gmail.com>
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Romainnorberg\Residue\Exception;
+
+use Exception;
```
Comment (romainnorberg) [inline src/Exception/DivideByZeroException.php]: unused

File: src/Exception/StepException.php [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+/*
+ * This file is part of the Residue package.
+ * (c) Romain Norberg <romainnorberg@gmail.com>
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Romainnorberg\Residue\Exception;
+
+use Exception;
```
Comment (romainnorberg) [inline src/Exception/StepException.php]: unused

**Comment by MockingMagician** [comment]
> Hy @MockingMagician Nice, thanks  . Could you rebase this with the current master ? I pushed some modifications _(including the static constructor)_ before seeing your PR

You're welcomed. :smile: It is rebased from your master branch.

**Comment by romainnorberg** [comment]
Thanks @MockingMagician

**Pull request merged**